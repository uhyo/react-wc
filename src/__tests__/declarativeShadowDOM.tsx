/**
 * @jest-environment node
 */
import React from "react";
import { renderToString } from "react-dom/server";
import { wc, wcIntrinsic } from "..";

describe("declarativeShadowDOM", () => {
  describe("wc", () => {
    it("one slot", () => {
      const Hello = wc({
        shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div><slot></slot></div>
      `,
        slots: [],
        name: "wc-test-foo",
        declarativeShadowDOM: true,
      });

      const str = renderToString(
        <Hello>
          <span>Foobar</span>
        </Hello>
      );

      expect(str).toMatchSnapshot();
    });
    it("named slot", () => {
      const Hello = wc({
        shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <slot name="foo"></slot>
      `,
        slots: ["foo"],
        name: "wc-test-2",
        declarativeShadowDOM: true,
      });

      const str = renderToString(<Hello foo={<span>Hi</span>} />);

      expect(str).toMatchSnapshot();
    });
  });
  describe("wcIntrinsic", () => {
    it("one slot", () => {
      const Hello = wcIntrinsic({
        shadowHtml: `
          <style>
            :host {
              color: red;
            }
          </style>
          <slot></slot>
      `,
        slots: [],
        element: "span",
        declarativeShadowDOM: true,
      });

      const str = renderToString(
        <Hello>
          <span>Foobar</span>
        </Hello>
      );

      expect(str).toMatchSnapshot();
    });
    it("two slots", () => {
      const Hello = wcIntrinsic({
        shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div><slot name="foo"></slot><slot name="bar"></slot></div>
      `,
        slots: ["foo", "bar"],
        element: "div",
        declarativeShadowDOM: true,
      });

      const str = renderToString(<Hello foo={<span>Hi</span>} bar="wow" />);

      expect(str).toMatchSnapshot();
    });
  });
});
