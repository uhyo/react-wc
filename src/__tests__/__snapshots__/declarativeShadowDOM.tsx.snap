// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declarativeShadowDOM wc one slot 1`] = `
"<wc-test-foo data-reactroot=\\"\\"><template shadowroot=\\"open\\">
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div><slot></slot></div>
      </template><span>Foobar</span></wc-test-foo>"
`;

exports[`declarativeShadowDOM wcIntrinsic two slots 1`] = `
"<div foo=\\"[object Object]\\" bar=\\"wow\\" data-reactroot=\\"\\"><template shadowroot=\\"open\\">
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div><slot name=\\"foo\\"></slot><slot name=\\"bar\\"></slot></div>
      </template><span slot=\\"foo\\">Hi</span><wc-text slot=\\"bar\\">wow</wc-text></div>"
`;
